### ISSUE TITLE
Regression for swipl-wasm

### ISSUE THREAD

**Issue by josd** [SWI-Prolog/swipl-devel#1007]
For the latest swipl-devel and using https://github.com/josd/eye/blob/master/eyejs/mkswipl-wasm we get some errors:
```
~/github.com/SWI-Prolog/swipl-devel$ mkswipl-wasm
~/github.com/SWI-Prolog/swipl-devel ~/github.com/SWI-Prolog/swipl-devel
Already up to date.
Adding directories to PATH:
PATH += /home/jdroo/emsdk
PATH += /home/jdroo/emsdk/upstream/emscripten
PATH += /home/jdroo/emsdk/node/14.18.2_64bit/bin

Setting environment variables:
PATH = /home/jdroo/emsdk:/home/jdroo/emsdk/upstream/emscripten:/home/jdroo/emsdk/node/14.18.2_64bit/bin:/home/jdroo/.elan/bin:/home/jdroo/.local/bin:/home/jdroo/.ciao/build/bin:/home/jdroo/github.com/ciao-lang/ciao/build/bin:/home/jdroo/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/wsl/lib:/mnt/c/ProgramFiles/WindowsApps/CanonicalGroupLimited.Ubuntu20.04onWindows_2004.2022.8.0_x64__79rhkp1fndgsc:/mnt/c/windows/system32:/mnt/c/windows:/mnt/c/windows/System32/Wbem:/mnt/c/windows/System32/WindowsPowerShell/v1.0/:/mnt/c/windows/System32/OpenSSH/:/mnt/c/ProgramFiles(x86)/NVIDIACorporation/PhysX/Common:/mnt/c/ProgramFiles/NVIDIACorporation/NVIDIANvDLISR:/mnt/c/WINDOWS/system32:/mnt/c/WINDOWS:/mnt/c/WINDOWS/System32/Wbem:/mnt/c/WINDOWS/System32/WindowsPowerShell/v1.0/:/mnt/c/WINDOWS/System32/OpenSSH/:/mnt/c/ProgramFiles/swipl/bin:/mnt/c/Users/josde/AppData/Local/Microsoft/WindowsApps:/mnt/c/ProgramFiles/VcXsrv:/mnt/c/Users/josde/AppData/Local/Programs/MicrosoftVSCode/bin:/mnt/c/ProgramFiles/eye/bin:/snap/bin:/home/jdroo/julia-1.6.3/bin
EMSDK = /home/jdroo/emsdk
EM_CONFIG = /home/jdroo/emsdk/.emscripten
EMSDK_NODE = /home/jdroo/emsdk/node/14.18.2_64bit/bin/node
Unknown C compiler.  Clang
-- Configuring SWI-Prolog-8.5.14
-- Check floating point format
-- Check floating point format - unknown
-- Found ZLIB: /home/jdroo/zlib-1.2.11/libz.a (found version "1.2.11")
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1")
-- Could NOT find LibTCMalloc (missing: LIBTCMALLOC_LIBRARY)
-- Using install prefix "/home/jdroo/emsdk/upstream/emscripten/cache/sysroot"
-- Checking for 64-bit off_t
-- Checking for 64-bit off_t - present
-- Checking for fseeko/ftello
-- Checking for fseeko/ftello - present
-- Large File support - found
-- Could NOT find LibTCMalloc (missing: LIBTCMALLOC_LIBRARY)
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for wchar.h
-- Looking for wchar.h - found
-- Looking for alloca.h
-- Looking for alloca.h - found
-- Looking for curses.h
-- Looking for curses.h - not found
-- Looking for dirent.h
-- Looking for dirent.h - found
-- Looking for dl.h
-- Looking for dl.h - not found
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for dmalloc.h
-- Looking for dmalloc.h - not found
-- Looking for execinfo.h
-- Looking for execinfo.h - not found
-- Looking for float.h
-- Looking for float.h - found
-- Looking for floatingpoint.h
-- Looking for floatingpoint.h - not found
-- Looking for ieeefp.h
-- Looking for ieeefp.h - not found
-- Looking for ieee754.h
-- Looking for ieee754.h - not found
-- Looking for libloaderapi.h
-- Looking for libloaderapi.h - not found
-- Looking for limits.h
-- Looking for limits.h - found
-- Looking for locale.h
-- Looking for locale.h - found
-- Looking for malloc.h
-- Looking for malloc.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Looking for ncurses/curses.h
-- Looking for ncurses/curses.h - not found
-- Looking for ncurses/term.h
-- Looking for ncurses/term.h - not found
-- Looking for poll.h
-- Looking for poll.h - found
-- Looking for pwd.h
-- Looking for pwd.h - found
-- Looking for shlobj.h
-- Looking for shlobj.h - not found
-- Looking for signal.h
-- Looking for signal.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for sys/dir.h
-- Looking for sys/dir.h - found
-- Looking for sys/file.h
-- Looking for sys/file.h - found
-- Looking for sys/mman.h
-- Looking for sys/mman.h - found
-- Looking for sys/ndir.h
-- Looking for sys/ndir.h - not found
-- Looking for sys/param.h
-- Looking for sys/param.h - found
-- Looking for sys/resource.h
-- Looking for sys/resource.h - found
-- Looking for sys/select.h
-- Looking for sys/select.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/syscall.h
-- Looking for sys/syscall.h - found
-- Looking for sys/termio.h
-- Looking for sys/termio.h - not found
-- Looking for sys/termios.h
-- Looking for sys/termios.h - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for sys/wait.h
-- Looking for sys/wait.h - found
-- Looking for term.h
-- Looking for term.h - not found
-- Looking for time.h
-- Looking for time.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for valgrind/valgrind.h
-- Looking for valgrind/valgrind.h - not found
-- Looking for vfork.h
-- Looking for vfork.h - not found
-- Looking for mach/thread_act.h
-- Looking for mach/thread_act.h - not found
-- Looking for sys/stropts.h
-- Looking for sys/stropts.h - found
-- Looking for zlib.h
-- Looking for zlib.h - found
-- Looking for crt_externs.h
-- Looking for crt_externs.h - not found
-- Looking for gperftools/malloc_extension_c.h
-- Looking for gperftools/malloc_extension_c.h - not found
-- Performing Test HAVE_GCC_ATOMIC
-- Performing Test HAVE_GCC_ATOMIC - Success
-- Performing Test HAVE_GCC_ATOMIC_8
-- Performing Test HAVE_GCC_ATOMIC_8 - Success
-- Looking for dlopen in dl
-- Looking for dlopen in dl - found
-- Looking for sin in m
-- Looking for sin in m - found
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of int
-- Check size of int - done
-- Check size of long
-- Check size of long - done
-- Check size of void *
-- Check size of void * - done
-- Check size of long long
-- Check size of long long - done
-- Check size of wchar_t
-- Check size of wchar_t - done
-- Check size of wint_t
-- Check size of wint_t - done
-- Check size of off_t
-- Check size of off_t - done
-- Looking for strerror
-- Looking for strerror - found
-- Looking for poll
-- Looking for poll - found
-- Looking for popen
-- Looking for popen - found
-- Looking for getpwnam
-- Looking for getpwnam - found
-- Looking for fork
-- Looking for fork - found
-- Looking for vfork
-- Looking for vfork - found
-- Looking for qsort_r
-- Looking for qsort_r - not found
-- Looking for qsort_s
-- Looking for qsort_s - not found
-- Looking for getpagesize
-- Looking for getpagesize - found
-- Looking for access
-- Looking for access - found
-- Looking for chmod
-- Looking for chmod - found
-- Looking for fchmod
-- Looking for fchmod - found
-- Looking for fcntl
-- Looking for fcntl - found
-- Looking for fstat
-- Looking for fstat - found
-- Looking for ftruncate
-- Looking for ftruncate - found
-- Looking for getcwd
-- Looking for getcwd - found
-- Looking for getwd
-- Looking for getwd - not found
-- Looking for opendir
-- Looking for opendir - found
-- Looking for readlink
-- Looking for readlink - found
-- Looking for remove
-- Looking for remove - found
-- Looking for rename
-- Looking for rename - found
-- Looking for stat
-- Looking for stat - found
-- Looking for fopen64
-- Looking for fopen64 - found
-- Looking for memmove
-- Looking for memmove - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for stricmp
-- Looking for stricmp - not found
-- Looking for strlwr
-- Looking for strlwr - found
-- Looking for setlocale
-- Looking for setlocale - found
-- Looking for mbsnrtowcs
-- Looking for mbsnrtowcs - found
-- Looking for mbcasescoll
-- Looking for mbcasescoll - not found
-- Looking for localeconv
-- Looking for localeconv - found
-- Looking for wcsdup
-- Looking for wcsdup - found
-- Looking for wcsxfrm
-- Looking for wcsxfrm - found
-- Looking for getuid
-- Looking for getuid - found
-- Looking for getpid
-- Looking for getpid - found
-- Looking for waitpid
-- Looking for waitpid - found
-- Looking for setenv
-- Looking for setenv - found
-- Looking for putenv
-- Looking for putenv - found
-- Looking for unsetenv
-- Looking for unsetenv - found
-- Looking for sysconf
-- Looking for sysconf - found
-- Looking for confstr
-- Looking for confstr - found
-- Looking for getrlimit
-- Looking for getrlimit - found
-- Looking for getrusage
-- Looking for getrusage - found
-- Looking for _NSGetEnviron
-- Looking for _NSGetEnviron - not found
-- Looking for mallinfo
-- Looking for mallinfo - found
-- Looking for mallinfo2
-- Looking for mallinfo2 - not found
-- Looking for shl_load
-- Looking for shl_load - not found
-- Looking for dlopen
-- Looking for dlopen - found
-- Looking for dladdr
-- Looking for dladdr - found
-- Looking for signal
-- Looking for signal - found
-- Looking for sigprocmask
-- Looking for sigprocmask - found
-- Looking for sigsetmask
-- Looking for sigsetmask - not found
-- Looking for siggetmask
-- Looking for siggetmask - not found
-- Looking for sigaction
-- Looking for sigaction - found
-- Looking for sigset
-- Looking for sigset - not found
-- Looking for sigblock
-- Looking for sigblock - not found
-- Looking for sigaltstack
-- Looking for sigaltstack - not found
-- Looking for kill
-- Looking for kill - found
-- Looking for backtrace
-- Looking for backtrace - not found
-- Looking for alarm
-- Looking for alarm - found
-- Looking for mtrace
-- Looking for mtrace - not found
-- Looking for tgetent
-- Looking for tgetent - not found
-- Looking for tcsetattr
-- Looking for tcsetattr - found
-- Looking for grantpt
-- Looking for grantpt - found
-- Looking for sgttyb
-- Looking for sgttyb - not found
-- Looking for cfmakeraw
-- Looking for cfmakeraw - found
-- Looking for ceil
-- Looking for ceil - found
-- Looking for floor
-- Looking for floor - found
-- Looking for srand
-- Looking for srand - found
-- Looking for srandom
-- Looking for srandom - found
-- Looking for random
-- Looking for random - found
-- Looking for rint
-- Looking for rint - found
-- Looking for fpclass
-- Looking for fpclass - not found
-- Looking for fpresetsticky
-- Looking for fpresetsticky - not found
-- Looking for fpsetmask
-- Looking for fpsetmask - not found
-- Looking for isnan
-- Looking for isnan - not found
-- Looking for isinf
-- Looking for isinf - not found
-- Looking for ftime
-- Looking for ftime - found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for gettimeofday
-- Looking for gettimeofday - found
-- Looking for localtime_r
-- Looking for localtime_r - found
-- Looking for localtime_s
-- Looking for localtime_s - not found
-- Looking for ctime_r
-- Looking for ctime_r - found
-- Looking for asctime_r
-- Looking for asctime_r - found
-- Looking for nanosleep
-- Looking for nanosleep - found
-- Looking for sleep
-- Looking for sleep - found
-- Looking for usleep
-- Looking for usleep - found
-- Looking for select
-- Looking for select - found
-- Looking for clock
-- Looking for clock - found
-- Looking for times
-- Looking for times - found
-- Looking for delay
-- Looking for delay - not found
-- Looking for dossleep
-- Looking for dossleep - not found
-- Looking for WSAPoll
-- Looking for WSAPoll - not found
-- Looking for WinExec
-- Looking for WinExec - not found
-- Looking for F_SETLKW
-- Looking for F_SETLKW - found
-- Performing Test HAVE_VAR_TIMEZONE
-- Performing Test HAVE_VAR_TIMEZONE - Success
-- Looking for SIGPROF
-- Looking for SIGPROF - found
-- Performing Test HAVE_STRUCT_TIME_TM_GMTOFF
-- Performing Test HAVE_STRUCT_TIME_TM_GMTOFF - Success
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM - Success
-- Performing Test HAVE_RU_IDRSS
-- Performing Test HAVE_RU_IDRSS - Success
-- Performing Test HAVE_GMP_RANDINIT_MT
-- Performing Test HAVE_GMP_RANDINIT_MT - Failed
-- Performing Test HAVE__BUILTIN_CLZ
-- Performing Test HAVE__BUILTIN_CLZ - Success
-- Performing Test HAVE__BUILTIN_POPCOUNT
-- Performing Test HAVE__BUILTIN_POPCOUNT - Success
-- Performing Test HAVE___BUILTIN_EXPECT
-- Performing Test HAVE___BUILTIN_EXPECT - Success
-- Performing Test HAVE___BUILTIN_CHOOSE_EXPR
-- Performing Test HAVE___BUILTIN_CHOOSE_EXPR - Success
-- Performing Test HAVE___BUILTIN_TYPES_COMPATIBLE_P
-- Performing Test HAVE___BUILTIN_TYPES_COMPATIBLE_P - Success
-- Performing Test HAVE___THREAD
-- Performing Test HAVE___THREAD - Success
-- Performing Test HAVE_VOLATILE
-- Performing Test HAVE_VOLATILE - Success
-- Performing Test HAVE_INLINE
-- Performing Test HAVE_INLINE - Success
-- Performing Test O_LABEL_ADDRESSES
-- Performing Test O_LABEL_ADDRESSES - Success
-- Performing Test HAVE_WEAK_ATTRIBUTE
-- Performing Test HAVE_WEAK_ATTRIBUTE - Success
-- Performing Test HAVE_VISIBILITY_ATTRIBUTE
-- Performing Test HAVE_VISIBILITY_ATTRIBUTE - Success
-- Performing Test HAVE_ALLOCA
-- Performing Test HAVE_ALLOCA - Failed
-- Performing Test HAVE_SIGNBIT
-- Performing Test HAVE_SIGNBIT - Success
-- Check for signal return type in <signal.h>
-- Check for signal handler return type type void  - found
-- sysconf() support for _SC_NPROCESSORS_CONF
-- sysconf() supports _SC_NPROCESSORS_CONF  - found
-- Performing Test TIME_WITH_SYS_TIME
-- Performing Test TIME_WITH_SYS_TIME - Success
-- llround(nextoward(0.5),-10) == 0 (ok)
-- copysign(1.0, modf(-0.0, _)) == -1.0 (ok)
CMake Warning (dev) at src/CMakeLists.txt:428 (add_library):
  ADD_LIBRARY called with SHARED option but the target platform does not
  support dynamic linking.  Building a STATIC library instead.  This may lead
  to problems.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Finding default package generator
-- Setup for packaging with DEB
-- Configuring done
-- Generating done
-- Build files have been written to: /home/jdroo/github.com/SWI-Prolog/swipl-devel/build.wasm
[196/366] Building C object src/CMakeFiles/swiplobjs.dir/pl-write.c.o
../src/pl-write.c:877:1: warning: unused function 'get_chr_from_text' [-Wunused-function]
get_chr_from_text(const PL_chars_t *t, int index)
^
1 warning generated.
[243/366] Building C object src/CMakeFiles/swiplobjs.dir/os/pl-cstack.c.o
FAILED: src/CMakeFiles/swiplobjs.dir/os/pl-cstack.c.o
/home/jdroo/emsdk/upstream/emscripten/emcc  -Isrc -I../src -I/home/jdroo/zlib-1.2.11 -O3 -DNDEBUG -fPIC -fvisibility=hidden   -Wall -std=gnu11 -MD -MT src/CMakeFiles/swiplobjs.dir/os/pl-cstack.c.o -MF src/CMakeFiles/swiplobjs.dir/os/pl-cstack.c.o.d -o src/CMakeFiles/swiplobjs.dir/os/pl-cstack.c.o   -c ../src/os/pl-cstack.c
../src/os/pl-cstack.c:1086:3: error: unknown type name 'PL_thread_info_t'; did you mean 'PL_thread_attr_t'?
{ PL_thread_info_t *info = LD->thread.info;
  ^~~~~~~~~~~~~~~~
  PL_thread_attr_t
../src/SWI-Prolog.h:1262:3: note: 'PL_thread_attr_t' declared here
} PL_thread_attr_t;
  ^
../src/os/pl-cstack.c:1086:32: error: no member named 'thread' in 'struct PL_local_data'
{ PL_thread_info_t *info = LD->thread.info;
                           ~~  ^
../src/os/pl-cstack.c:1088:14: error: no member named 'c_stack_size' in 'PL_thread_attr_t'
  if ( info->c_stack_size )
       ~~~~  ^
../src/os/pl-cstack.c:1089:18: error: no member named 'c_stack_size' in 'PL_thread_attr_t'
    return info->c_stack_size;
           ~~~~  ^
../src/os/pl-cstack.c:1116:11: error: no member named 'c_stack_size' in 'PL_thread_attr_t'
    info->c_stack_size = rlim.rlim_cur;
    ~~~~  ^
../src/os/pl-cstack.c:1118:11: error: no member named 'c_stack_base' in 'PL_thread_attr_t'
    info->c_stack_base = (void*)(top - info->c_stack_size);
    ~~~~  ^
../src/os/pl-cstack.c:1118:46: error: no member named 'c_stack_size' in 'PL_thread_attr_t'
    info->c_stack_base = (void*)(top - info->c_stack_size);
                                       ~~~~  ^
../src/os/pl-cstack.c:1120:11: error: no member named 'c_stack_size' in 'PL_thread_attr_t'
  { info->c_stack_size = (size_t)-1;
    ~~~~  ^
../src/os/pl-cstack.c:1124:16: error: no member named 'c_stack_size' in 'PL_thread_attr_t'
  return info->c_stack_size;
         ~~~~  ^
9 errors generated.
emcc: error: '/home/jdroo/emsdk/upstream/bin/clang -target wasm32-unknown-emscripten -DEMSCRIPTEN -D__EMSCRIPTEN_major__=3 -D__EMSCRIPTEN_minor__=1 -D__EMSCRIPTEN_tiny__=4 -fignore-exceptions -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr -D_LIBCPP_ABI_VERSION=2 -fno-threadsafe-statics -Werror=implicit-function-declaration -Xclang -iwithsysroot/include/SDL --sysroot=/home/jdroo/emsdk/upstream/emscripten/cache/sysroot -Xclang -iwithsysroot/include/compat -Isrc -I../src -I/home/jdroo/zlib-1.2.11 -O3 -DNDEBUG -fPIC -fvisibility=hidden -Wall -std=gnu11 -MD -MT src/CMakeFiles/swiplobjs.dir/os/pl-cstack.c.o -MF src/CMakeFiles/swiplobjs.dir/os/pl-cstack.c.o.d -c ../src/os/pl-cstack.c -o src/CMakeFiles/swiplobjs.dir/os/pl-cstack.c.o' failed (returned 1)
[252/366] Building C object src/CMakeFiles/swiplobjs.dir/pl-wam.c.o
ninja: build stopped: subcommand failed.
```

**Comment by JanWielemaker** [comment]
Thanks.  Fixed with SWI-Prolog/swipl-devel@842f541537bed4a50cf7d095d9b61e373eb74330

**Comment by josd** [comment]
Thanks @JanWielemaker and it is now working fine for https://github.com/josd/eye/tree/master/eyejs
